docs/tutorials/zero-to-hero.md

```diff
@@ -308,601 +116,78 @@ export function getResponseByKey(key: string): string {
-> - Add a button to invoke the method (and optionally render the result to the console or div)
-
-Hints:
-- string --> number `I32.parseInt(numberAsText)`
-- number --> string `someNumber.toString()`
-- at the time of writing, one should avoid using `i32()` and friends for typecasting strings to numbers
-
-
-How did it go?
-
-You can see how we did it here...
-
-> In the file `assembly/main.ts`
-> - We appended the code below to the bottom of the file
-
-```ts
-export function finalizeBet(): void {
-  let price = storage.getString("btc-price");
-
-  // expecting price as string in the format #,###.####
-  price = price.split(',').join('').split('.')[0];
-
-  if (price) {
-    let btcPrice: i32 = I32.parseInt(price);
-    let overUnderPrice: i32 = 6000;
-    let outcome: string;
-
-    if (btcPrice > overUnderPrice) {
-      outcome = "BTC price is " + btcPrice.toString() + " - Pay Moon Hodler 2000 USD"
-    } else {
-      outcome = "BTC price is " + btcPrice.toString() + " - Pay FUD Hodler 2000 USD"
-    }
-
-    storage.setString("betOutcome", outcome)
-
-  } else {
-    storage.setString("betOutcome", "btc price is undefined")
-  }
+  //stripping only the data we want from the API response
+  let data = body.bpi.USD.rate
+  //Saving the data to the blockchain by calling the Oracle Contracts setResponse function
+  await window.contract.setResponse({ apiResponse: data });
+  // Check to see if the data was saved properly
+  let apiResponse = await contract.getResponse();
+  console.log(`${apiResponse} is the API response available to the Oracle on-chain`);
 }
 ```
 
-> In the file `src/main.js`
-> - We surfaced the new contract method `finalizeBet` to our JavaScript contract
-
+Then replace the event handler of the button with the following:
 ```js
-window.contract = await near.loadContract(nearConfig.contractName, {
-  viewMethods: ["getResponse", "getResponseByKey", "getOracleQueryParams"],
-  // find the line below and change it to match
-  changeMethods: ["setResponse", "setResponseByKey", "setOracleQueryParams", "finalizeBet"],
-  sender: window.accountId
-});
-```
-
-> Also in the same file `src/main.js`
-> - We appended the code below to the bottom of the file
-
-```js
-async function finalizeBet() {
-  await contract.finalizeBet()
-  let outcome = await contract.getResponseByKey({ key: "betOutcome" })
-  console.log(outcome)
-  document.getElementById("betOutcome").innerText = outcome
-}
-```
-
-> In the file `src/index.html`
-> - We appended the code below just before the closing tag of `div.after-sign-in`
-
-```html
-<button onClick={finalizeBet()}>Run Wager Smart Contract Code + Print Outcome</button>
-<div class="container" style="background-color:blue; color:white;">
-  The Bet Outcome is:
-  <div id="betOutcome"></div>
-</div>
+  document.getElementById('set-response').addEventListener('click', () => {
+    makeApiCallAndSave();
+  });
 ```
+This should save the price of Bitcoin to the blockchain.
```

Comment:
This saves the price of Bitcoin to the blockchain.

Reply:
done